wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth1/OAuthAuthenticationService.java

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.security.oauth1;
+
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.rest.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.eclipse.che.commons.lang.UrlUtils.getParameter;
+import static org.eclipse.che.commons.lang.UrlUtils.getQueryParametersFromState;
+import static org.eclipse.che.commons.lang.UrlUtils.getRequestUrl;
+import static org.eclipse.che.commons.lang.UrlUtils.getState;
+
+/**
+ * RESTful wrapper for OAuth 1.0.
+ *
+ * @author Kevin Pollet
+ * @author Igor Vinokur
+ */
+@Path("oauth/1.0")
+public class OAuthAuthenticationService extends Service {
+    private static final Logger LOG = LoggerFactory.getLogger(OAuthAuthenticationService.class);
+
+    private final static String USER_ID_PARAMETER                = "user_id";
+    private final static String REQUEST_URL_PARAMETER            = "request_url";
+    private final static String PROVIDER_NAME_PARAMETER          = "oauth_provider";
+    private final static String REQUEST_METHOD_PARAMETER         = "request_method";
+    private final static String SIGNATURE_METHOD_PARAMETER       = "signature_method";
+    private final static String REDIRECT_AFTER_LOGIN_PARAMETER   = "redirect_after_login";
+    private final static String UNSUPPORTED_OAUTH_PROVIDER_ERROR = "Unsupported OAuth provider: %s";
+
+    @Inject
+    protected OAuthAuthenticatorProvider providers;
+
+    @GET
+    @Path("authenticate")
+    public Response authenticate(@Context UriInfo uriInfo) throws OAuthAuthenticationException, BadRequestException {
+        MultivaluedMap<String, String> parameters = uriInfo.getQueryParameters();
```

Comment:
Why using `uriInfo` here is better than injection parameters by `@QueryParam` annotation?